app/main.py

```diff
@@ -1,13 +1,45 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors":
+            [
+                {
+                    "line": key["line_number"],
+                    "column": key["column_number"],
+                    "message": key["text"],
+                    "name": key["code"],
+                    "source": "flake8"
+                }
+                for key in errors
+            ],
+        "path": file_path,
+        "status": "failed" if len(errors) > 1 else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "path": keys,
```

Comment:
use function `format_single_linter_file ` to format a single file. It allows you to avoid code duplication